File: Algorithms/Graphs/ShortestPaths/dijkstra.py
from Queue import PriorityQueue

MAXN = 10 # MAX NODE

adj = {i: [] for i in range(MAXN)}
dist = {i: float('INF') for i in range(MAXN)}

def edge(f, t, w):
    adj[f].append((t, w))

def Dijkstra(node):
    pq = vPriorityQueue()
    dist[node] = 0
    pq.put((0, node))
    while not pq.empty():
        v = pq.get()[1]
        for u in adj[v]:
            node, weight = u[0], u[1]
            if dist[v] + weight < dist[node]:
                dist[node] = dist[v] + weight
                pq.put((dist[node], node))


# from, to, weight
edge(1, 5, 1)

File: Algorithms/Graphs/shortest_path/dijkstra.cpp
#include <iostream>
#include <queue>
#include <vector>

using namespace std;

const int MAXN = 10;
const int INF = 1e7;
vector<vector<pair<int, int>>> adj(MAXN);

int dijkstra(int src, int dest)
{
    queue<pair<int, int>> buf;
    vector<int> dist(MAXN, INF);
    dist[src] = 0;
    buf.push({0, src});
    while (!buf.empty())
    {
        int v = buf.front().second;
        buf.pop();
        for (auto u : adj[v])
        {
            int node = u.first;
            int weight = u.second;
            if (dist[v] + weight < dist[node])
            {
                dist[node] = dist[v] + weight;
                buf.push({dist[node], node});
            }
        }
    }
    return dist[dest];
}

int main()
{
    int E;
    cin >> E;
    int F, T, W;
    while (E--)
    {
        cin >> F >> T >> W;
        adj[F].push_back({T, W});
        adj[T].push_back({F, W});
    }
    cout << dijkstra(1, 6) << endl;
}